src/main/java/com/ibm/cloud/sdk/core/util/CredentialUtils.java

```diff
@@ -462,4 +400,149 @@ private static ServiceCredentials setCredentialFields(String serviceName, List<S
 
     return serviceCredentials;
   }
+
+  /**
+   * Returns a Map containing properties found within the credential file that are associated with the
+   * specified cloud service.
+   * @param serviceName the name of the cloud service whose properties should be loaded
+   * @return a Map containing the properties
+   */
+  public static Map<String, String> getFileCredentialsAsMap(String serviceName) {
```

Comment:
It looks like there's actually a lot more we can clean up in this file.
- `getFileCredentialsAsMap()` is almost identical to the old `getFileCredentials()`, other than the return type. I think we can just get rid of the old method.
- `getCredentialsFromVcap()` is also pretty redundant with the addition of `getVcapCredentialsAsMap()`. I think we can refactor the latter method to just try and get the individual fields directly. Something like:

    ```java
    public static Map<String, String> getVcapCredentialsAsMap(String serviceName) {
      Map<String, String> props = new HashMap<>();
      addToMap(props, Authenticator.PROPNAME_USERNAME, getVcapValue(serviceName, Authenticator.PROPNAME_USERNAME));
      addToMap(props, Authenticator.PROPNAME_PASSWORD, getVcapValue(serviceName, Authenticator.PROPNAME_PASSWORD));
      // etc...
    }
    ```
- With the above change, we can probably also just get rid of the `ServiceCredentials` inner class.
- The whole `setCredentialFields()` method can also be deleted in favor of the new `parseCredentials()`.

Reply:
I was a little hesitant to remove all the "old" code from CredentialUtils as I wasn't sure how much (if any) we might have exposed to users.   But if you think that's not an issue, I'm happy to carve it up even more :)